cheatsheets/Protect_FileUpload_Against_Malicious_File.md

```diff
@@ -1,597 +1,138 @@
-                        //Handle only formats for which Apache Commons Imaging can successfully write 
-                        // (YES in Write column of the reference link) the image format
-                        //See reference link in the class header
-                        switch (formatName) {
-                            case "TIFF": {
-                                imageParser = new TiffImageParser();
-                                break;
-                            }
-                            case "PCX": {
-                                imageParser = new PcxImageParser();
-                                break;
-                            }
-                            case "DCX": {
-                                imageParser = new DcxImageParser();
-                                break;
-                            }
-                            case "BMP": {
-                                imageParser = new BmpImageParser();
-                                break;
-                            }
-                            case "GIF": {
-                                imageParser = new GifImageParser();
-                                break;
-                            }
-                            case "PNG": {
-                                imageParser = new PngImageParser();
-                                break;
-                            }
-                            case "WBMP": {
-                                imageParser = new WbmpImageParser();
-                                break;
-                            }
-                            case "XBM": {
-                                imageParser = new XbmImageParser();
-                                break;
-                            }
-                            case "XPM": {
-                                imageParser = new XpmImageParser();
-                                break;
-                            }
-                            default: {
-                                throw new IOException("Format of the original image is not" + 
-                                                      " supported for write operation !");
-                            }
-
-                        }
-                        imageParser.writeImage(sanitizedImage, fos, new HashMap<>());
-                    }
-
-                }
-
-                // Set state flag
-                safeState = true;
-            }
-        } catch (Exception e) {
-            safeState = false;
-            LOG.warn("Error during Image file processing !", e);
-        }
-
-        return safeState;
-    }
-}
+```python
+# This code snippet only serves as an example, and should not be used
+blacklist = ["php", "php5"]  # can be replaced with a regex pattern, but that would be even weaker, such as \.php$
+for bad in blacklist:
+    if "file.php".endswith(bad):
+        // raise exception
 ```
 
-# Sources of the prototype
+### User Permissions
```

Comment:
Section seems a bit out of place - file type and content type are much more related. I'd either move this earlier or alter.